src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String[] actions = { "supply", "buy" };
+    private final StringBuilder text = new StringBuilder();
+
+    public void readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            if (text.length() == 0) {
+                String readByLine = bufferedReader.readLine();
+
+                while (readByLine != null) {
+                    text.append(readByLine).append(",");
+                    readByLine = bufferedReader.readLine();
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file", e);
+        }
+    }
 
+    public void writeToTheFile(String toFileName) {
+        String[] split = text.toString().split(",");
+        int[] amount = new int[actions.length];
+
+        for (int i = 0; i < split.length; i++) {
+            for (int j = 0; j < actions.length; j++) {
+                if (actions[j].equals(split[i])) {
+                    amount[j] += Integer.parseInt(split[i + 1]);
+                }
+            }
+        }
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(actions[0] + "," + amount[0] + System.lineSeparator()
+                        + actions[1] + "," + amount[1] + System.lineSeparator()
+                        + "result" + "," + (amount[0] - amount[1]));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to the file", e);
+        }
+    }
+
+    public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        writeToTheFile(toFileName);
```

Comment:
`readFromFile` should return result of read operation, `writeToTheFile` should accept data with params. After reading and before writing you have to make `calculations`.